src/azul/drs.py

```diff
@@ -74,10 +84,95 @@ class Access:
     headers: Optional[Mapping[str, str]] = None
 
 
+@attr.s(auto_attribs=True, kw_only=True, frozen=True, slots=True)
+class DRSCompactPseudoURI:
+    """
+    So-called DRS "URIs" [1] for Compact Identifiers [2] are NOT URIs according
+    to RFC 3986 [3] so we can't use off-the-shelf URI parsers.
+
+    [1] https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.3.0/docs/
+
+    [2] https://www.nature.com/articles/sdata201829
+
+    [3] https://datatracker.ietf.org/doc/html/rfc3986
+    """
+    scheme: str
+    netloc: str
+    prefix: str
+    object_id: str
+
+    @classmethod
+    def parse(cls, drs_uri: str) -> 'DRSCompactPseudoURI':
+        scheme, netloc = drs_uri.split('://', 1)
+        prefix, object_id = netloc.split(':', 1)
+        return cls(scheme=scheme,
+                   netloc=netloc,
+                   prefix=prefix,
+                   object_id=object_id)
+
+    @classmethod
+    def maybe_parse(cls, drs_uri: str) -> Union[mutable_furl, 'DRSCompactPseudoURI']:
```

Comment:
I think we should use a proper class hierarchy for this. DRSURI as the base class, CompactDRSURI (we'll drop the pseudo bit) and RegularDRSURI as the concrete subclasses. This method would become DRSURI.parse.